src/daemon/rpmostreed-daemon.cxx

```diff
@@ -737,10 +737,21 @@ char *
 rpmostreed_daemon_client_get_sd_unit (RpmostreedDaemon *self, const char *client)
 {
   auto clientdata = static_cast<struct RpmOstreeClient *>(g_hash_table_lookup (self->bus_clients, client));
-  if (!clientdata || clientdata->sd_unit == NULL)
-    return NULL;
-  else
-    return g_strdup (clientdata->sd_unit);
+  /* If the caller didn't register (e.g. Zincati doesn't today because it runs as a
+   * separate UID which systemd doesn't consider `active`), then let's look up its
+   * systemd unit now.
+   */
+  if (!clientdata)
+    {
+      pid_t pid;
+      char *sd_unit = NULL;
```

Comment:
Doesn't this leak? `sd_pid_get_user_unit(3)` says:

> The returned string needs to be freed with the libc free(3) call after use.

So I think instead you can just return that string and avoid the `g_strdup` below.

Reply:
ah right. I should've checked the man page. sorry!